src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1322,16 +1344,43 @@ private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pA
             }
             else
             {
-                if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                if (Sse2.IsSupported)
                 {
-                    return 0;
+                    if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                    {
+                        return 0;
+                    }
+                }
+                else if (AdvSimd.Arm64.IsSupported)
+                {
+                    if (ContainsNonAsciiValue(AdvSimd.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate)))
```

Comment:
If I understand the original Sse2 algorithm right - this is testing if any of the characters in `utf16VectorFirst` is outside of `[0, 0x7f]` value range, right?

I don't think the algorithm on Arm64 should copy exactly, what Sse2 does. 

I believe the following should be enough
```c#
AdvSimd.MaxPairwise(utf16VectorFirst.AsUInt16(), utf16VectorFirst.AsUInt16()).AsUInt64().ToScalar() & 0xFF80FF80FF80FF80UL != 0
```

Reply:
I agree.